### PR TITLE
chore(nns): reduce logging in tests 

### PR DESCRIPTION
**Pull request by max-dfinity**
There are many logs that are helpful in mainnet that are creating too much noise in tests, causing in some cases the testing infrastructure to get overwhelmed and hang during cleanup.  

Additionally, removing the extra logs reduces the test run time from ~40s to  ~5s.

### CODE REVIEW

File: rs/nervous_system/neurons_fund/src/polynomial_matching_function_tests.rs [hunk 1]
```diff
@@ -55,6 +55,7 @@ fn polynomial_matching_function_viability_test() {
         let f = assert_matches!(PolynomialMatchingFunction::new(
             *total_maturity_equivalent_icp_e8s,
             neurons_fund_participation_limits,
+            true
```
Comment (max-dfinity) [inline rs/nervous_system/neurons_fund/src/polynomial_matching_function_tests.rs:58]: true in this test, b/c it's a test directly about the thing

**Review by anchpop — APPROVED** [review]

File: rs/nns/governance/src/neurons_fund.rs [hunk 2]
```diff
@@ -1075,26 +1075,32 @@ where
...
-                    "{}WARNING: Neurons' Fund has zero total maturity.",
-                    governance::LOG_PREFIX
-                );
+                #[cfg(not(test))]
+                {
+                    println!(
+                        "{}WARNING: Neurons' Fund has zero total maturity.",
+                        governance::LOG_PREFIX
+                    );
+                }
```
Comment (jasonz-dfinity) [inline rs/nns/governance/src/neurons_fund.rs:1084]: There is no `use dfn_core::println` at the top? If so, this actually means the logs are not printed in production.

I believe `use dfn_core::println` achieves roughly the same thing as what's done here. In unit tests (not compiled as wasm), those will not be printed, while in state machine tests, pocket ic tests or production, those will be printed. It's possible that we want to disable the logging in state machine tests and pocket ic tests, but I believe the `#[cfg(not(test))]` blocks are also enabled in those tests.

File: rs/nns/governance/src/neurons_fund.rs [hunk 3]
```diff
@@ -2417,7 +2428,7 @@ mod test_functions_tests {
                 let f = LinearFunction { slope, intercept };
                 for i in generate_potentially_intresting_target_values() {
                     let target_y = u64_to_dec(i).unwrap();
-                    println!("Inverting linear function {target_y} = f(x) = {slope} * x + {intercept} ...");
+                    // println!("Inverting linear function {target_y} = f(x) = {slope} * x + {intercept} ...");
```
Comment (aterga) [inline rs/nns/governance/src/neurons_fund.rs:2431]: This is probably not intended to stay, right?

File: rs/nns/governance/src/neurons_fund.rs (hunk 2 above)
Comment (max-dfinity) [reply rs/nns/governance/src/neurons_fund.rs:1084]: Yeah this is a good point.  

Maybe the logs should just be removed instead?

Comment (aterga) [reply rs/nns/governance/src/neurons_fund.rs:1084]: (It's easy to forget to set up an NF in testing, and then spend time understanding why it didn't participate in an SNS)

Comment (aterga) [reply rs/nns/governance/src/neurons_fund.rs:1084]:
```suggestion
```

**Review by aterga — APPROVED** [review]

**Review by daniel-wong-dfinity-org — COMMENTED** [review]
NCR

In general, I do not like dropping data. Ofc, I also hate log spam. These concerns are somewhat in tension, but I think we can satisfy both. In general, we need to be _selective_. Throwing out the bath water good. Throwing out the baby bad.

I wouldn't say all the deleted logging are super high value, but they aren't spam either.

I don't think the title matches. More precisely, some of the deleted logging is NOT only during tests. I like what the title claims more than what the code changes actually do.

Ideally (i.e. not in this PR), we would have a generalized ability to filter out logs that are not needed in tests, rather than deleting logging entirely. At Google there is this flag called [--vmodule](https://chromium.googlesource.com/playground/chromium-org-site/+/refs/heads/main/for-testers/enable-logging/index.md), which is kind of what I'm thinking of. You could use that to turn on "debug" level logs by filename.

**Comment by anchpop** [comment]
> NCR
> 
> In general, I do not like dropping data. Ofc, I also hate log spam. These concerns are somewhat in conflict, but I think we can usually satisfy both. The key is to be _selective_. Throwing out the bath water good. Throwing out the baby bad.
> 
> I wouldn't say all the deleted logging are super high value, but they aren't spam either.
> 
> I don't think the title matches. More precisely, some of the deleted logging is NOT only during tests. I like what the title claims more than what the code changes actually do.
> 
> Ideally (i.e. maybe not in this PR), we would have a generalized ability to filter out logs (and apply that to tests), rather than deleting logging entirely. At Google, there is this nice flag called [--vmodule](https://chromium.googlesource.com/playground/chromium-org-site/+/refs/heads/main/for-testers/enable-logging/index.md), which is kind of what I'm thinking of. You could use that to turn on "super debug" level logs by filename. Maybe what we want here is the opposite: be able to turn OFF some logs.
> 
> Maybe, we could do that like this:
> 
> ```rust
> #[cfg(???)]
> use filter_out_some_logs_in_tests::println;
> ```

I agree that some of these are useful and they're not just being disabled in tests. But fast tests > logs IMO. I wonder if it would be possible to buffer them to get the best of both worlds.

**Comment by max-dfinity** [comment]
If it was just a case of fast tests or good production logs, it would be a bit more nuanced.

These logs were not printing in production, and some were only meant for tests in the first place, and they all created so much noise in tests that it broke our local test runners.  

We all agree logs are good.

**Pull request merged**